### PR TITLE
Replace blacklist with blocklist, whitelist with allowlist

### PR DESCRIPTION
**Pull request by mercedesb**
This PR replaces all instances of blacklist with blocklist and whitelist with allowlist. None of this terminology is user-facing or exposed via the API therefore the changes are backward compatible. This feels like a small change to make the codebase more welcoming to contributors 🤗 

Let me know if you'd like me to open an issue to propose this change rather than directly opening a PR.

## Rationale

Language matters and shapes our culture. Casually assigning positive connotations to white and negative connotations to black does not create an inclusive culture. Blocklist and allowlist are also less ambiguous terms for people unfamiliar with their concepts. Blocklist and allowlist are a more accurate representation of what they're used for.

### CODE REVIEW

**Comment by simi** [comment]
If doing change in here, I think we can use more descriptive terms. In this case it seems majority of naming is related to reserved gems. Can we rename methods to similar naming? `reserved?`, ...

https://github.com/rubygems/rfcs/pull/40#discussion_r854511802

**Comment by mercedesb** [comment]
> If doing change in here, I think we can use more descriptive terms. In this case it seems majority of naming is related to reserved gems. Can we rename methods to similar naming? `reserved?`, ...
> 
> [rubygems/rfcs#40 (comment)](https://github.com/rubygems/rfcs/pull/40#discussion_r854511802)

Absolutely! I'll go ahead and take care of that.

File: config/initializers/forbidden_yaml.rb [hunk 1]
```diff
@@ -9,7 +9,7 @@
 
 module Gem
   class Specification
-    WHITELISTED_CLASSES = %w(
+    ALLOWLISTED_CLASSES = %w(
```
Comment (simi) [inline config/initializers/forbidden_yaml.rb:12]: `ALLOWED_CLASSES` would be enough here

File: test/unit/links_test.rb [hunk 2]
```diff
@@ -56,7 +56,7 @@ class LinksTest < ActiveSupport::TestCase
     assert links.homepage_uri
   end
 
-  context "metadata includes non whitelisted uri key" do
+  context "metadata includes non allowlisted uri key" do
```
Comment (simi) [inline test/unit/links_test.rb:59]: "metadata includes non allowlisted uri key"

File: test/unit/links_test.rb [hunk 3]
```diff
@@ -67,23 +67,23 @@ class LinksTest < ActiveSupport::TestCase
         "funding_uri"         => "https://example.com",
         "documentation_uri"   => "https://example.com",
         "changelog_uri"       => "https://example.com",
-        "non_whitelisted_uri" => "https://example.com"
+        "non_allowlisted_uri" => "https://example.com"
```
Comment (simi) [inline test/unit/links_test.rb:70]: "unknown_uri"

File: test/unit/links_test.rb [hunk 4]
```diff
@@ -67,23 +67,23 @@ class LinksTest < ActiveSupport::TestCase
...
 
       version = build(:version, metadata: metadata)
       rubygem = build(:rubygem, versions: [version])
       @links = rubygem.links(version)
     end
 
-    should "create method for whitelisted keys" do
-      whitelisted_keys = Links::LINKS.values.reject! { |k| k == "download_uri" }
-      whitelisted_keys.each do |key|
+    should "create method for allowlisted keys" do
```
Comment (simi) [inline test/unit/links_test.rb:78]: "create method for known keys"

File: test/unit/links_test.rb [hunk 5]
```diff
@@ -67,23 +67,23 @@ class LinksTest < ActiveSupport::TestCase
...
+      allowlisted_keys = Links::LINKS.values.reject! { |k| k == "download_uri" }
+      allowlisted_keys.each do |key|
         assert_equal "https://example.com", @links.send(key), "value doesn't match for method: #{key}"
       end
     end
 
-    should "not create method for non whitelisted key" do
-      refute_respond_to @links, "non_whitelisted_uri"
+    should "not create method for non allowlisted key" do
```
Comment (simi) [inline test/unit/links_test.rb:85]: "not create method for unknown key"

**Pull request merged**